src/fromager/commands/graph.py

```diff
@@ -153,6 +162,69 @@ def explain_duplicates(wkctx, graph_file):
             print(f"  * No single version of {dep_name} meets all requirements")
 
 
+@graph.command()
+@click.option(
+    "--version",
+    type=clickext.PackageVersion(),
+    multiple=True,
+    help="filter by version for the given package",
+)
+@click.option(
+    "--recursive-depth",
+    type=int,
+    default=0,
+    help="recursively get why each package depends on each other. Set depth to -1 for full recursion till root",
+)
+@click.option(
+    "--req-type",
+    type=clickext.RequirementType(),
+    multiple=True,
+    help="filter by req type",
+)
+@click.argument(
+    "graph-file",
+    type=clickext.ClickPath(),
+)
+@click.argument("package-name", type=str)
+@click.pass_obj
+def why(
+    wkctx: context.WorkContext,
+    graph_file: pathlib.Path,
+    package_name: str,
+    version: list[Version],
+    recursive_depth: int,
+    req_type: list[RequirementType],
+):
+    "Explain why a dependency shows up in the graph"
+    graph = DependencyGraph.from_file(graph_file)
+    package_nodes = graph.get_nodes_by_name(package_name)
+    if version:
+        package_nodes = [node for node in package_nodes if node.version in version]
+    for node in package_nodes:
+        print(f"\n{node.key}")
+        find_why(graph, node, recursive_depth, 1, req_type)
+
+
+def find_why(
+    graph: DependencyGraph,
+    node: DependencyNode,
+    recursive_depth: int,
+    depth: int,
+    req_type: list[RequirementType],
+):
+    for parent in node.parents:
+        if parent.destination_node.key == ROOT:
+            continue
+        if req_type and parent.req_type not in req_type:
+            continue
+
+        print(
+            f"{'  ' * depth} * is an {parent.req_type} dependency of {parent.destination_node.key} with req {parent.req}"
+        )
+        if recursive_depth and (recursive_depth == -1 or depth <= recursive_depth):
+            find_why(graph, parent.destination_node, recursive_depth, depth + 1, [])
```

Comment:
If I filter based on requirement types I might not see any output of this loop. A message saying that all dependencies were filtered would help understand that case.